deluge/core/alertmanager.py

```diff
@@ -41,13 +41,43 @@
 
 """
 
-from twisted.internet import reactor
+import threading
+
+from twisted.internet import reactor, threads
 
 import deluge.component as component
 from deluge._libtorrent import lt
 
 from deluge.log import LOG as log
 
+class AlertPoller(object):
```

Comment:
Typically you would subclass threading.Thread and override run() for something like this.  However, I think that you could probably even forego the class entirely and just run a thread of a method in AlertManager itself.  You could probably use the state of the Component class in lieu of self.running. I would just write a new method called 'wait_for_alerts' (essentially your run() method) and run that as a thread.  It will have access to all the member variables of AlertManager and you don't need to worry about synchronization since Python has the GIL.

Here is an example of just running a method as a thread:

```python
import threading


class AlertManager:
    def __init__(self):
        self.foo = 'bar'

    def thread(self):
        if self.foo == 'bar':
            self.foo = 'foobar'

    def start(self):
        print(self.foo)
        t = threading.Thread(target=self.thread)
        t.start()
        t.join()
        print(self.foo)

a = AlertManager()
a.start()
```

Reply:
Done.